src/citrine/_rest/collection.py

```diff
@@ -39,15 +40,37 @@ def _put_resource_ref(self, subpath: str, uid: Union[UUID, str]):
     def _get_path(self,
                   uid: Optional[Union[UUID, str]] = None,
                   *,
-                  ignore_dataset: Optional[bool] = False) -> str:
-        """Construct a url from __base_path__ and, optionally, id."""
-        subpath = format_escaped_url('/{}', uid) if uid else ''
+                  ignore_dataset: bool = False,
+                  action: Optional[Sequence] = None,
+                  query_terms: Optional[Dict[str, str]] = None,
+                  ) -> str:
+        """Construct a url from __base_path__ and, optionally, id and/or action."""
         if ignore_dataset:
-            return format_escaped_url(self._dataset_agnostic_path_template + subpath,
-                                      **self.__dict__)
+            base = urlparse(self._dataset_agnostic_path_template)
         else:
-            return format_escaped_url(self._path_template + subpath,
-                                      **self.__dict__)
+            base = urlparse(self._path_template)
+        path = base.path.split('/')
+
+        if uid is not None:
+            path.append("{uid}")
+
+        if action is None:
+            action = []
+        elif isinstance(action, str):
+            action = [action]
+        else:
+            action = list(action)
+        path.extend(["{}"] * len(action))
+
+        if query_terms is None:
+            query_terms = {}
+        query_args = [x for k_and_v in query_terms.items() for x in k_and_v]
```

Comment:
Instead of this and your join below, you can use `urllib.parse.urlencode`: [https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode)

Reply:
Makes sense, though it'd be slightly better if I moved that into a argument to `format_escaped_url` -- we're mixing responsibilities.

Done.